### PR TITLE
Word card branch

### PR DESCRIPTION
**Pull request by serpient**
gif

## Things Done
- Changed card state management to ENUMs because we are only rendering 1 state at a time (we change only the border colors when state changes). 
- Fixed card match functions
- Add resetGame function
- Add shuffle card function
- hook up lifecycle functions to handle onMount and whenever props change from parent (aka, changing the dataset language)

## Things To Do
- add instructional text instead of placeholder text @evaplaysviolin 
- style the buttons that changes dataset @evaplaysviolin 
- add form input for user-generated datasets!! This will be a nice meaty thing to practice  React/Javascript with! @evaplaysviolin or @kellyburg 
- use localStorage to save game state and user-generated dataset @AnutaK . Right now, refreshing mid-way through a game would completely reset the component. Can you save every change made within the game until the "reset" button is clicked?
- get this thing deployed! @AnutaK

### CODE REVIEW

**Comment by evaplaysviolin** [comment]
Hello! I'm still looking through what you've written to understand it better (just for my learning purposes) but things seem to be working for the most part.

I'd like to know why you added the scale on hover? I personally think it's kind of distracting and makes the animations look less smooth. That aside, though, the cards still scale when they should be deactivated (or am I imagining that?).

Also, this isn't related to what you've done - but I was wondering if anyone feels a way about the longer text not being centered? I can't remember if I did that on purpose but it looks kind of weird to me now...should I change that?

**Comment by serpient** [comment]
@evaplaysviolin  i added scale onHover as a personal preference -- just felt like it needed some sort of non-static interaction to make the cards stand out more. you can try something else instead of scale!

yes to centering long text!

**Comment by evaplaysviolin** [comment]
I also have a question about where the language set buttons are rendering from. Should we be organizing the renders differently (for instance, everything else right now renders inside of VocabGamePlay, including things like the header, which isn't related to actual game play)? Where should the (S)CSS for these buttons go?

**Comment by evaplaysviolin** [comment]
@serpient Re: scaling

I had a border that showed up on hover already (or is that static)? Perhaps too subtle?

**Comment by serpient** [comment]
possibly too subtle, but not a big deal.

the language set buttons are rendering from App.JSX

File: src/App.js [hunk 1]
```diff
@@ -1,11 +1,46 @@
...
+    let { name } = e.currentTarget;
+    this.setState({ currentDataset: GameDataset[name], currentDatasetName: name });
+  }
+
   render() {
+    let { currentDataset } = this.state;
     return (
       <div className="App">
-        <VocabGamePlay />
+        { this.renderVocabButtons() }
```
Comment (serpient) [inline src/App.js:42]: vocab buttons are here!
its rendering above the vocab game play page

Comment (evaplaysviolin) [reply src/App.js:42]: No I know that, I'm asking about organization lol

File: src/VocabGameData.js [hunk 2]
```diff
@@ -35,26 +68,30 @@ export function createDataset(base) {
...
+  japanese: japanesePairs,
+}
 try {
-  GameDataset = createDataset(basePairs)
+  for (var key in GameDataset) {
+    GameDataset[key] = createDataset(GameDataset[key])
+  }
 } catch(error) {
   console.error(error.message);
   GameDataset = new Map();
```
Comment (serpient) [inline src/VocabGameData.js:97]: delete this line -- unused

Comment (serpient) [reply src/VocabGameData.js:97]: line 97**

**Comment by serpient** [comment]
if we want to use one page for everything (which is the direction we are
heading :D ) , yeah we should move TITLE and game description onto App.jsx

and then just have a App.scss that takes care of styling for those
components and the vocab buttons

**Comment by evaplaysviolin** [comment]
Okay, thanks for understanding my question lol! My review is that things look fine for pushing but we'll still have to adjust some stuff after, like:

- Moving stuff to render into App
- Re-doing the relevant CSS for above changes/move to new file

I can do that (unless someone else wants to), but would prefer to merge into master first?

And then of course before we do that, we should have @kellyburg and @AnutaK review, too. Does that sound good?

**Comment by serpient** [comment]
sounds good!
and yeah lets merge into master first before commencing more work :D

**Pull request merged**